nltk/embedding/word2vec.py

```diff
@@ -0,0 +1,134 @@
+        """
+        :param sentences: list of sentences for training 
+        :type sentences: iter(list(str)) 
+        :param sg: defines the training algorithm. By default (sg=1), skip-gram is used. Otherwise, cbow is employed.
+        :param min_count:  ignore all words with total frequency lower than this.
+        :param size:  is the dimensionality of the feature vectors.
+        :param window:  is the maximum distance between the current and predicted word within a sentence.
+        :param thread: Number of thread for training. Noteed that Cython is needed for multiple thread training. 
+        :param hs: if 1 (default), hierarchical sampling will be used for model training (else set to 0).
+        :param negative: if > 0, negative sampling will be used, the int for negative specifies how many "noise words" should be drawn (usually between 5-20). If negative > 0, hs must set to 0.  
+        :param iteration: number of iterations (epochs) over the corpus.
+        """
+        _model = gensim.models.Word2Vec(sentences,min_count= min_count, size = size, workers = thread, window = window,
+                                       hs = hs, negative = negative, iter = iteration, sg=sg)
+        self.model = _model;
+
+    def _check_model(self):
+        if self.model is None : 
+            raise ValueError(" Need to train or load the model ")
+        
+    def save_model(self, file_name):
+        """
+        Save the model using pickle
+        """ 
+        self._check_model()
+        self.model.save(file_name)
+        
+    def load_model(self,file_name):
+        """
+        Load the saved model
+        """ 
+        self.model = gensim.models.Word2Vec.load(file_name)
+
+    def get_vector(self, word):
+        """
+        :return : Get the vector representation of the word. None if the word is missing from the model.  
+        """ 
+        self._check_model()
+        
+        try:
+            return self.model[word]
+        except KeyError:
+            return None   
+    
+    def similarity(self,word1,word2):
+        """
+        :return: the cosine similarity of two words. -1 if any word is missing from the model.   
+        """ 
+        self._check_model()
+        
+        try:
+            return self.model.similarity(word1,word2);
+        except KeyError:
+            return -1  
+        
+    def most_similar(self, positive=[], negative=[], topn=10):
+        """
+        :param positive: the list of positive words contribute positively towards the similarity.
+        :param negative: the list of negative words contribute negatively toward the similarity. 
+        :param topn: return n result with the score.   
+        """
+        self._check_model()
+        try:
+            return self.model.most_similar(positive, negative, topn)
+        except KeyError:
+            return None 
+    
+        
+def demo():
+    ####### DEMO TRAINING ####### #######
```

Comment:
Hi @dimazest, 
Probably I should convert the part using pre-trained model to the doctest. I assume that for each training, the vector representation might be slightly different.